packages/ui/src/elements/WhereBuilder/Condition/Relationship/index.tsx

```diff
@@ -209,7 +205,9 @@ export const RelationshipField: React.FC<Props> = (props) => {
   }, [hasMany, hasMultipleRelations, value, options])
 
   const handleInputChange = (input: string) => {
+    dispatchOptions({ type: 'CLEAR', i18n, required: false })
     const relationSlug = partiallyLoadedRelationshipSlugs.current[0]
+    partiallyLoadedRelationshipSlugs.current = relationSlugs
```

Comment:
Both refs are reset to the initial values ​​of the component.
If this line is not added, when scrolling to a last page, nothing else can be searched again.